### ISSUE TITLE
[QUESTION] Is there a command for showing remote content?

### ISSUE THREAD

**Issue by techmuch** [wshanks/tbkeys#136]
For example, I would like a hotkey to show images within an html email

**Comment by trulycool** [comment]
I would also love to know if there is a command for this. 

When first installing tbkeys-lite a year or two back, could not track it down. Ended up having to hack together an AutoHotkey shortcut that sends the literal keypresses, so when I use Alt + C in mail reader window it sends the key sequence exactly as if you were pressing Alt + O (opens options menu from button) then S (selects and runs the Show Remote Content entry).

It works, but feels hacky compared to knowing the TB internal command. And of course it's Windows only, since afaik there is no AHK for Mac or Linux...

Here is my AHK code in case it's helpful to others until someone sorts out the internal TB command:

```
;### ThunderbirdDone fixes
; Show Remote Contect shortcut to Thunderbird
#IfWinActive ahk_exe thunderbird.exe
	; alt + c
	!c:: ; Thunderbird Show Remote Content
	Send {Alt down}o{Alt up}s
return
;### END: Thunderbird fixes
```

**Comment by trulycool** [comment]
It appears that once upon a time, TB had an internal command to do this:
> LoadMsgWithRemoteContent
...however sadly, this does not appear to work w/ current TB versions :(

**Comment by morat523035** [comment]
The following command works with the error console.

```
window.LoadMsgWithRemoteContent();
```

Are you using tbkey or tbkeys-lite?

You cannot run arbitrary javascript with tbkeys-lite.

There is a tbkeys.xpi link on the GitHub releases page.

**Comment by trulycool** [comment]
Well would you look at that, it works great in console!

I am currently running tbkeys-lite, I ended up w/ that since it was what was available in Thunderbird Add-ons.

I guess they not allow you to list the regular tbkeys there because of the ability to run arbitrary javascript w/ eval? Frustrating...

I wrote a somewhat extended plea on the TB keybinding API request thread you mention in the tbkeys repo.

Thank you for all your hard work on this extension, and for taking the time to respond to this thread. Much appreciated 🙏

**Comment by trulycool** [comment]
Thank you @morat523035 ! 

While I hack about Firefox & FF Extensions, rarely mess w/ Thunderbird (tho it is my main mail app). Hadn't even tried the Dev Tools console...

I just assumed that TB had dropped the command altogether. But after seeing it run via console...was inspired to push a bit more...

And just after writing a [probably way too long] impassioned plea to the Thunderbird dev thread where @wshanks points in the full tbkeys repo, wherein he succinctly explains how his extension and the wonderful Nostalgy were both hobbled by the removal of a proper keyboard API...

I went back to the tbkeys-lite options, and noticed that in addition to the `cmd:` prefix, which calls internal TB commands, and which did not work w/ LoadMsgWithRemoteContent, there is a `func:` prefix too...

**And voila! We can haz keyboard shortcut to 'Show remote content in message' w/ tbkeys-lite!**

```
"ctrl+shift+c": "func:LoadMsgWithRemoteContent"
```

Thanks to all for the code and info; open source rules!

🙏🙌🕸

**Comment by techmuch** [comment]
Works perfectly!!! 🙏 🙏 🙏

**Comment by techmuch** [comment]
Thank you for your hard work and dedication to open-source software it is greatly appreciated. Thank you for making email management even easier!

**Comment by Crandel** [comment]
Since thunderbird 115 update `func:LoadMsgWithRemoteContent` is not working anymore

**Comment by morat523035** [comment]
The LoadMsgWithRemoteContent function is now defined in the aboutMessage.xhtml window, not the messenger.xhtml window.

Try the following command with tbkeys.

```
window.gTabmail.currentAboutMessage.LoadMsgWithRemoteContent();
```

You cannot run arbitrary javascript with tbkeys-lite.

About new mail front end
http://developer.thunderbird.net/thunderbird-development/codebase-overview/mail-front-end
http://developer.thunderbird.net/add-ons/updating/tb115/adapt-to-changes-in-thunderbird-103-115

**Comment by Crandel** [comment]
I'm using tbkeys, but have no idea how to run this code.
 This is my settings and nothing happens when I press Ctrl+h.

```"ctrl+h": "(function () {window.gTabmail.currentAboutMessage.LoadMsgWithRemoteContent(); window.content.focus();});"```

**Comment by Crandel** [comment]
Thank you. Now everything is working correctly

**Comment by wshanks** [comment]
> `"ctrl+h": "(function () {window.gTabmail.currentAboutMessage.LoadMsgWithRemoteContent(); window.content.focus();});"`

Here I think the issue might have been a missing `(` -- you have an unmatched `)` at the end. I think the end should be `()`.

**Comment by schonda** [comment]
did anyone bring this to work on thunderbird 128.6 with tbkeys-lite 2.3??

since this would be a really useful function hopefully mozilla developers can implement it in some future release...

**Comment by Crandel** [comment]
Looks like `window.gTabmail.currentAboutMessage.LoadMsgWithRemoteContent();` not working anymore. Can somebody find working solution?

**Comment by morat523035** [comment]
@Crandel

What app version are you running?

The code snippet works for me in Thunderbird 140. I did not test the code snippet in Thunderbird 149.

Releases
http://www.thunderbird.net/thunderbird/releases/

The function LoadMsgWithRemoteContent is still available in Thunderbird 151.

Reference
http://searchfox.org/comm-central/search?q=LoadMsgWithRemoteContent
http://searchfox.org/comm-central/source/mail/config/version.txt

Try testing the code snippet using the error console.

i.e. Menu Bar > Tools > Developer Tools > Error Console

Does the `window.gTabmail.currentAboutMessage;` property result in the `ChromeWindow about:message` object using the error console?

**Comment by Crandel** [comment]
Yes, I saw in source code, but `console.log(window.gTabmail.currentAboutMessage)` show me `null`. Looks like this was a xul function and they remove this function from gTabmail together with `currentAbout3Pane`. This is on fresh install. The same version on home laptop is still working and it was created years ago

**Comment by morat523035** [comment]
@Crandel

What app version are you running?

e.g. Thunderbird Release 148 or Thunderbird Release 149 or Thunderbird Beta 150

Did you select a message? Did you enable the message pane?

```
// null when message isn't selected or message pane is disabled in Thunderbird ESR 140
console.log(window.gTabmail.currentAboutMessage);
```

i.e. Menu Bar > View > Layout > [check] Message Pane (F8)

About Mail Front End
http://developer.thunderbird.net/thunderbird-development/codebase-overview/mail-front-end
http://developer.thunderbird.net/add-ons/updating/tb115/adapt-to-changes-in-thunderbird-103-115

Reference (comm-esr140 is not available)
http://searchfox.org/comm-central/search?q=currentAboutMessage
http://searchfox.org/comm-esr128/search?q=currentAboutMessage

> they remove this function from gTabmail together with currentAbout3Pane

That's a huge change if true. The addon developer for ImportExportTools NG is still using `gTabmail.currentAbout3Pane` and `gTabmail.currentAboutMessage` in his latest beta.

Try running the following code snippet using the error console.

```
(function () {
  var tabmail = window.document.getElementById('tabmail');
  console.log(tabmail.currentAbout3Pane);
  console.log(tabmail.currentAboutMessage);
})();
```

ImportExportTools NG
http://addons.thunderbird.net/thunderbird/addon/986686
http://github.com/thunderbird/import-export-tools-ng/issues/856

**Comment by Crandel** [comment]
I'm using 149.0.2 versions on Arch Linux. BTW on my home instance, where `currentAboutMessage` is available I use import-export-tools addon, but on working Arch Linux in the same version I don't. Maybe it is working only if this addon is active?

>Did you select a message? Did you disable the message pane?

I'm selecting the message.
How to disable message pane? Maybe I did it accidentally?

**Comment by morat523035** [comment]
@Crandel

> How to disable message pane?

I can press F8 to toggle the message pane in Thunderbird 140.

Thunderbird keyboard shortcuts
http://support.mozilla.org/kb/keyboard-shortcuts-thunderbird

**Comment by Crandel** [comment]
> I can press F8 to toggle the message pane in Thunderbird 140.

Now I understand, what you mean. No, I've never disabled message pane at all.

**Comment by morat523035** [comment]
@Crandel

The code snippet works for me in Thunderbird 149. I am using table view message list display and classic view layout.

```
console.log(window.gTabmail.currentAbout3Pane);   // ChromeWindow about:3pane
console.log(window.gTabmail.currentAboutMessage); // ChromeWindow about:message
```

```
(function () {
  var tabmail = window.document.getElementById('tabmail');
  console.log(tabmail.currentAbout3Pane);   // ChromeWindow about:3pane
  console.log(tabmail.currentAboutMessage); // ChromeWindow about:message
})();
```

```
window.gTabmail.currentAboutMessage.LoadMsgWithRemoteContent();
```

Thunderbird Portable 149.0.2 (fresh install)
Windows 10 Pro 22H2 32-bit

Thunderbird Portable
http://portableapps.com/apps/internet/thunderbird_portable

I like to troubleshoot using a portable fresh install by adding a newsgroup account with a fake name and fake email address to show "Local Folders" in the folder pane, then opening a saved message, then copying the saved message to a new folder in Local Folders.

Your Name: `John Smith` ; Email Address: `user@example.net`

Newsgroup server: `news.mozilla.org` ; Account Name: `Mozilla News` ; Newsgroup: `mozilla.announce`

I can still create the newsgroup account, even though Mozilla's mailman service has been decommissioned.
